agent/xdsv2/rbac_resources.go

```diff
@@ -169,6 +168,197 @@ func makeRBACPolicy(p *pbproxystate.Permission) *envoy_rbac_v3.Policy {
 	}
 }
 
+func l4PolicyLabel(perms []*pbproxystate.Permission, i int) string {
+	if len(perms) == 1 {
+		return baseL4PermissionKey
+	}
+	return fmt.Sprintf("%s-%d", baseL4PermissionKey, i)
+}
+
+func makeL7RBACPolicies(l7Permissions []*pbproxystate.Permission) map[string]*envoy_rbac_v3.Policy {
+	// sort permissions into those with L7-specific features and those without, to match labeling and behavior
+	// conventions in V1: https://github.com/hashicorp/consul/blob/4e451f23584473a7eaf7f123145ca85e0a31783a/agent/xds/rbac.go#L647
+	// this is a somewhat unfortunate carry-over needed for testing v1 vs v2 final config
+	// and this will break with v1 intentions when multiple L4 permissions are used
```

Comment:
Based on the linked code, it looks like it is just processing L4 and L7 intentions differently. Is the l4/l7 ordering happening somewhere else?